misk-hibernate/src/main/kotlin/misk/hibernate/Transacter.kt

```diff
@@ -13,11 +13,26 @@ interface Transacter {
    * Starts a transaction on the current thread, executes lambda, and commits the transaction.
    * If lambda raises an exception the transaction will be rolled back instead of committed.
    *
+   * If retries are permitted (the default), a failed but recoverable transaction will be
+   * reattempted after rolling back.
+   *
    * It is an error to start a transaction if another transaction is already in progress.
    */
   fun <T> transaction(lambda: (session: Session) -> T): T
+
+  fun retries(): Transacter
```

Comment:
why would I call this?

Reply:
It provides an option to turn retries back on. Eg. you pass around a configured transacter (rather than injecting a "default" one) and want to ensure that retries are enabled.